FullStoryMiddleware/FullStoryMiddleware.m

```diff
@@ -117,11 +134,28 @@ - (SEGPayload *)getNewPayloadWithFSURL:(SEGContext * _Nonnull)context {
             NSMutableDictionary *newProps = [[NSMutableDictionary alloc] initWithDictionary:screenPayload.properties];
             [newProps setValue:[FS currentSessionURL] forKey:@"FSSessionURL"];
 
-            newPayload = [[SEGScreenPayload alloc]
-                                           initWithName:screenPayload.name
-                                           properties:newProps
-                                           context:screenPayload.context
-                                           integrations:screenPayload.integrations];
+            {
+                SEGScreenPayload *uninitPayload = [SEGScreenPayload alloc];
+
+                // init has a different signiture for Analytics version < 4
+                if ([uninitPayload respondsToSelector:@selector(initWithName:properties:context:integrations:)]) {
+                    newPayload = [uninitPayload
+                                  initWithName:screenPayload.name
+                                  properties:newProps
+                                  context:screenPayload.context
+                                  integrations:screenPayload.integrations];
+                } else if ([uninitPayload respondsToSelector:@selector(initWithName:category:properties:context:integrations:)]) {
+                    newPayload = [uninitPayload
+                                  initWithName:screenPayload.name
+                                  category:screenPayload.category
+                                  properties:newProps
+                                  context:screenPayload.context
+                                  integrations:screenPayload.integrations];
```

Comment:
This isn't what I meant by the alignment. I meant having the `:`s aligned, like you see in the `@interface`.

Reply:
Oops! I really want to add some linting to this project. But didn't seem like there's an official tool, would Clang be a good one to consider?